### DISCUSSION TITLE
How to build msbuild with mono?

### DISCUSSION THREAD

**Discussion by Kreijstal** [dotnet/msbuild#11172] [Q&A]
xbuild recommends to use msbuild, how to compile msbuild from mono?

**Comment by huoyaoyuan** [comment]
MSBuild uses MSBuild. To build MSBuild, you first need a existing version of MSBuild. You can execute the existing binaries on mono.

Using mono itself for building is not recommended either. You can use .NET Core for better support.

**Reply by Kreijstal** [reply]
ah no my goal was too bootstrap msbuild using open source tools, I guess the only way is to find the archive of an old msbuild version and build from there, failing that, to rewrite msbuild.
